chainer/functions/loss/softmax_cross_entropy.py

```diff
@@ -48,6 +54,12 @@ def forward_cpu(self, inputs):
         log_y = log_softmax._log_softmax(x, self.use_cudnn)
         if self.cache_score:
             self.y = numpy.exp(log_y)
+        if self.class_weight is not None:
+            if self.class_weight.shape != x.shape:
+                shape = [1 if d != 1 else -1 for d in six.moves.range(x.ndim)]
+                self.class_weight = numpy.broadcast_to(
+                    self.class_weight.reshape(shape), x.shape)
+            log_y *= self.class_weight
```

Comment:
I think that it's not particularly a problem because (i) the computational cost of `shape` is much lower than `log_y *=`, and (ii) both `broadcast_to` functions in `numpy` and `cupy` do not return an actual array but a readonly view.